diff -uNr xorgproto-2021.5/meson.build xorgproto-2021.5.mod/meson.build
--- xorgproto-2021.5/meson.build	2021-09-15 15:57:52.000000000 +0000
+++ xorgproto-2021.5.mod/meson.build	2021-09-22 02:55:00.876456101 +0000
@@ -62,7 +62,7 @@
     configure_file(
         input : pc + '.pc.in',
         output : pc + '.pc',
-        install_dir : get_option('datadir') + '/pkgconfig',
+        install_dir : '@TERMUX_PREFIX@/lib/pkgconfig',
         configuration : pc_data,
     )
 endforeach
@@ -100,7 +100,7 @@
         configure_file(
             input : pc + '.pc.in',
             output : pc + '.pc',
-            install_dir : get_option('datadir') + '/pkgconfig',
+            install_dir : '@TERMUX_PREFIX@/lib/pkgconfig',
             configuration : pc_data,
         )
     endforeach
