diff -uNr mobile/make/autoconf/toolchain.m4 mobile.mod/make/autoconf/toolchain.m4
--- mobile/make/autoconf/toolchain.m4	2021-06-10 08:31:10.623585305 +0000
+++ mobile.mod/make/autoconf/toolchain.m4	2021-06-10 08:40:56.686032281 +0000
@@ -444,22 +444,12 @@
     COMPILER_VERSION_OUTPUT=`$COMPILER --version 2>&1`
     # Check that this is likely to be clang
     $ECHO "$COMPILER_VERSION_OUTPUT" | $GREP "clang" > /dev/null
-    if test $? -ne 0; then
-      AC_MSG_NOTICE([The $COMPILER_NAME compiler (located as $COMPILER) does not seem to be the required $TOOLCHAIN_TYPE compiler.])
-      AC_MSG_NOTICE([The result from running with --version was: "$COMPILER_VERSION_OUTPUT"])
-      AC_MSG_ERROR([A $TOOLCHAIN_TYPE compiler is required. Try setting --with-tools-dir.])
-    fi
-    # Collapse compiler output into a single line
-    COMPILER_VERSION_STRING=`$ECHO $COMPILER_VERSION_OUTPUT`
-    COMPILER_VERSION_NUMBER=`$ECHO $COMPILER_VERSION_OUTPUT | \
-        $SED -e 's/^.* version \(@<:@1-9@:>@@<:@0-9.@:>@*\).*$/\1/'`
-  else
-      AC_MSG_ERROR([Unknown toolchain type $TOOLCHAIN_TYPE.])
-  fi
+
+    # Collapse compiler output into a single lin
   # This sets CC_VERSION_NUMBER or CXX_VERSION_NUMBER. (This comment is a grep marker)
-  $1_VERSION_NUMBER="$COMPILER_VERSION_NUMBER"
+  $1_VERSION_NUMBER="10.0.0-4"
   # This sets CC_VERSION_STRING or CXX_VERSION_STRING. (This comment is a grep marker)
-  $1_VERSION_STRING="$COMPILER_VERSION_STRING"
+  $1_VERSION_STRING="ubuntu1"
 
   AC_MSG_NOTICE([Using $TOOLCHAIN_TYPE $COMPILER_NAME compiler version $COMPILER_VERSION_NUMBER @<:@$COMPILER_VERSION_STRING@:>@])
 ])
