diff -uNr alsa-lib-1.1.3/configure alsa-lib-1.1.3.mod/configure
--- alsa-lib-1.1.3/configure	2016-12-20 14:43:55.000000000 +0000
+++ alsa-lib-1.1.3.mod/configure	2021-06-06 08:25:26.598511647 +0000
@@ -11978,14 +11978,6 @@
 
 test "x$prefix" = xNONE && prefix=$ac_default_prefix
 
-if test "$enable_static" = "$enable_shared" -a "$enable_static" = "yes"; then
-cat <<EOF
-  Please, do not try to compile static and shared libraries together.
-  See INSTALL file for more details (do not use --enable-shared=yes with
-  --enable-static=yes).
-EOF
-  exit 1
-fi
 
 
 # Check whether --with-configdir was given.
diff -uNr alsa-lib-1.1.3/include/mixer.h alsa-lib-1.1.3.mod/include/mixer.h
--- alsa-lib-1.1.3/include/mixer.h	2016-12-20 14:43:20.000000000 +0000
+++ alsa-lib-1.1.3.mod/include/mixer.h	2021-06-06 08:37:27.734595892 +0000
@@ -209,10 +209,12 @@
 	enum snd_mixer_selem_regopt_abstract abstract;
 	/** v1: device name (must be NULL when playback_pcm or capture_pcm != NULL) */
 	const char *device;
+	#ifndef __ANDROID__
 	/** v1: playback PCM connected to mixer device (NULL == none) */
 	snd_pcm_t *playback_pcm;
 	/** v1: capture PCM connected to mixer device (NULL == none) */
 	snd_pcm_t *capture_pcm;
+	#endif
 };
 
 /** Mixer simple element identifier */
diff -uNr alsa-lib-1.1.3/src/conf.c alsa-lib-1.1.3.mod/src/conf.c
--- alsa-lib-1.1.3/src/conf.c	2016-12-20 14:43:20.000000000 +0000
+++ alsa-lib-1.1.3.mod/src/conf.c	2021-06-06 08:47:26.192627794 +0000
@@ -3713,7 +3713,7 @@
 		if (S_ISDIR(st.st_mode)) {
 			struct dirent **namelist;
 			int n;
-
+#ifndef __ANDROID__
 #ifndef DOC_HIDDEN
 #if defined(_GNU_SOURCE) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__sun)
 #define SORTFUNC	versionsort
@@ -3722,6 +3722,7 @@
 #endif
 #endif
 			n = scandir(fi[idx].name, &namelist, config_filename_filter, SORTFUNC);
+#endif
 			if (n > 0) {
 				int j;
 				err = 0;
diff -uNr alsa-lib-1.1.3/src/ucm/parser.c alsa-lib-1.1.3.mod/src/ucm/parser.c
--- alsa-lib-1.1.3/src/ucm/parser.c	2016-12-20 14:43:20.000000000 +0000
+++ alsa-lib-1.1.3.mod/src/ucm/parser.c	2021-06-06 08:44:15.071969203 +0000
@@ -1418,13 +1418,15 @@
 	snprintf(filename, sizeof(filename)-1,
 		"%s", env ? env : ALSA_USE_CASE_DIR);
 	filename[MAX_FILE-1] = '\0';
-
+#ifndef __ANDROID__
 #if defined(_GNU_SOURCE) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__sun)
 #define SORTFUNC	versionsort
 #else
 #define SORTFUNC	alphasort
 #endif
+
 	err = scandir(filename, &namelist, filename_filter, SORTFUNC);
+#endif
 	if (err < 0) {
 		err = -errno;
 		uc_error("error: could not scan directory %s: %s",
