diff -uNr systemd/meson.build systemd.mod/meson.build
--- systemd/meson.build	2021-06-24 11:12:26.446012749 +0000
+++ systemd.mod/meson.build	2021-06-24 12:01:59.768662822 +0000
@@ -910,7 +910,6 @@
 #####################################################################
 
 threads = dependency('threads')
-librt = cc.find_library('rt')
 libm = cc.find_library('m')
 libdl = cc.find_library('dl')
 libcrypt = cc.find_library('crypt')
@@ -932,8 +931,6 @@
         libcap = cc.find_library('cap')
 endif
 
-libmount = dependency('mount',
-                      version : fuzzer_build ? '>= 0' : '>= 2.30')
 
 want_libfdisk = get_option('fdisk')
 if want_libfdisk != 'false' and not skip_deps
@@ -1663,7 +1660,6 @@
                      libbasic_gcrypt],
         link_whole : [libsystemd_static],
         dependencies : [threads,
-                        librt,
                         libxz,
                         libzstd,
                         liblz4],
@@ -1684,13 +1680,11 @@
         install_dir : rootlibdir,
         pic : static_libsystemd_pic,
         dependencies : [threads,
-                        librt,
                         libxz,
                         libzstd,
                         liblz4,
                         libcap,
                         libblkid,
-                        libmount,
                         libselinux,
                         libgcrypt],
         c_args : libsystemd_c_args + (static_libsystemd_pic ? [] : ['-fno-PIC']))
@@ -1722,8 +1716,8 @@
         install : static_libudev != 'false',
         install_dir : rootlibdir,
         link_depends : libudev_sym,
-        dependencies : libshared_deps + [libmount],
-        c_args : static_libudev_pic ? [] : ['-fno-PIC'],
+        dependencies : libshared_deps,
+	c_args : static_libudev_pic ? [] : ['-fno-PIC'],
         pic : static_libudev_pic)
 
 ############################################################
@@ -1821,8 +1815,7 @@
                         link_with : [libsystemd_static,
                                      libshared_static,
                                      libbasic],
-                        dependencies : [threads,
-                                        librt],
+                        dependencies : [threads],
                         link_depends : sym,
                         install : true,
                         install_dir : rootlibdir)
@@ -1852,10 +1845,8 @@
                      libshared],
         dependencies : [versiondep,
                         threads,
-                        librt,
                         libseccomp,
                         libselinux,
-                        libmount,
                         libblkid],
         install_rpath : rootlibexecdir,
         install : true,
@@ -1873,10 +1864,8 @@
                      libshared],
         dependencies : [versiondep,
                         threads,
-                        librt,
                         libseccomp,
                         libselinux,
-                        libmount,
                         libblkid],
         install_rpath : rootlibexecdir,
         install : conf.get('ENABLE_ANALYZE'))
@@ -3030,7 +3019,6 @@
         'src/mount/mount-tool.c',
         include_directories : includes,
         link_with : [libshared],
-        dependencies: [libmount],
         install_rpath : rootlibexecdir,
         install : true)
 
@@ -3190,7 +3178,6 @@
         systemd_shutdown_sources,
         include_directories : includes,
         link_with : [libshared],
-        dependencies : [libmount],
         install_rpath : rootlibexecdir,
         install : true,
         install_dir : rootlibexecdir)
diff -uNr systemd/src/core/meson.build systemd.mod/src/core/meson.build
--- systemd/src/core/meson.build	2021-06-24 11:12:26.507018382 +0000
+++ systemd.mod/src/core/meson.build	2021-06-24 11:50:13.000378132 +0000
@@ -150,14 +150,12 @@
         include_directories : includes,
         dependencies : [versiondep,
                         threads,
-                        librt,
                         libseccomp,
                         libpam,
                         libaudit,
                         libkmod,
                         libapparmor,
                         libselinux,
-                        libmount,
                         libacl])
 
 core_includes = [includes, include_directories('.')]
@@ -209,6 +207,5 @@
 fuzzers += [
         [['src/core/fuzz-unit-file.c'],
          [libcore,
-          libshared],
-         [libmount]],
+          libshared]],
 ]
diff -uNr systemd/src/libsystemd/meson.build systemd.mod/src/libsystemd/meson.build
--- systemd/src/libsystemd/meson.build	2021-06-24 11:12:26.544021800 +0000
+++ systemd.mod/src/libsystemd/meson.build	2021-06-24 11:30:14.312650235 +0000
@@ -166,8 +166,7 @@
         install : false,
         include_directories : libsystemd_includes,
         link_with : libbasic,
-        dependencies : [threads,
-                        librt],
+        dependencies : [threads],
         c_args : libsystemd_c_args)
 
 libsystemd_sym = files('libsystemd.sym')
diff -uNr systemd/src/shared/meson.build systemd.mod/src/shared/meson.build
--- systemd/src/shared/meson.build	2021-06-24 11:12:26.649031497 +0000
+++ systemd.mod/src/shared/meson.build	2021-06-24 11:34:33.133560225 +0000
@@ -384,11 +384,9 @@
                   libiptc,
                   libkmod,
                   liblz4,
-                  libmount,
                   libopenssl,
                   libp11kit,
                   libpam,
-                  librt,
                   libseccomp,
                   libselinux,
                   libzstd,
diff -uNr systemd/src/shutdown/meson.build systemd.mod/src/shutdown/meson.build
--- systemd/src/shutdown/meson.build	2021-06-24 11:12:26.660032513 +0000
+++ systemd.mod/src/shutdown/meson.build	2021-06-24 11:55:52.958779921 +0000
@@ -6,10 +6,6 @@
         umount.h
 '''.split())
 
-tests += [
-        [['src/shutdown/test-umount.c',
+tests +=[['src/shutdown/test-umount.c',
           'src/shutdown/umount.c',
-          'src/shutdown/umount.h'],
-         [],
-         [libmount]],
-]
+          'src/shutdown/umount.h']]
diff -uNr systemd/src/test/meson.build systemd.mod/src/test/meson.build
--- systemd/src/test/meson.build	2021-06-24 11:12:26.673033714 +0000
+++ systemd.mod/src/test/meson.build	2021-06-24 11:54:46.951682879 +0000
@@ -48,10 +48,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes],
 
@@ -73,10 +71,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes],
 
@@ -84,10 +80,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes, '', 'manual'],
 
@@ -95,10 +89,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes],
 
@@ -113,10 +105,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes],
 
@@ -124,10 +114,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes],
 
@@ -174,8 +162,7 @@
 
         [['src/test/test-libmount.c'],
          [],
-         [threads,
-          libmount]],
+         [threads]],
 
         [['src/test/test-mount-util.c']],
 
@@ -302,9 +289,7 @@
         [['src/test/test-bpf-devices.c'],
          [libcore,
           libshared],
-         [libmount,
-          threads,
-          librt,
+         [threads,
           libseccomp,
           libselinux,
           libblkid],
@@ -313,9 +298,7 @@
         [['src/test/test-bpf-firewall.c'],
          [libcore,
           libshared],
-         [libmount,
-          threads,
-          librt,
+         [threads,
           libseccomp,
           libselinux,
           libblkid],
@@ -324,9 +307,7 @@
         [['src/test/test-watch-pid.c'],
          [libcore,
           libshared],
-         [libmount,
-          threads,
-          librt,
+         [threads,
           libseccomp,
           libselinux,
           libblkid],
@@ -363,7 +344,6 @@
           libshared],
          [threads,
           libseccomp,
-          libmount,
           libxz,
           liblz4,
           libblkid],
@@ -430,10 +410,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes],
 
@@ -464,10 +442,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes, '', 'timeout=120'],
 
@@ -475,10 +451,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes, '', 'timeout=360'],
 
@@ -498,10 +472,8 @@
          [libcore,
           libshared],
          [threads,
-          librt,
           libseccomp,
           libselinux,
-          libmount,
           libblkid],
          core_includes],
 
@@ -524,7 +496,6 @@
          [libudevd_core,
           libshared],
          [threads,
-          librt,
           libblkid,
           libkmod,
           libacl,
