diff -uNr mesa/meson.build mesa.mod/meson.build
--- mesa/meson.build	2021-09-24 03:12:53.059805311 +0000
+++ mesa.mod/meson.build	2021-09-24 03:13:47.292284436 +0000
@@ -1057,14 +1057,7 @@
   endforeach
 else
   _trial = [
-    '-Werror=implicit-function-declaration',
-    '-Werror=missing-prototypes',
-    '-Werror=return-type',
-    '-Werror=empty-body',
-    '-Werror=incompatible-pointer-types',
-    '-Werror=int-conversion',
     '-Wimplicit-fallthrough',
-    '-Werror=thread-safety',
     '-Wno-missing-field-initializers',
     '-Wno-format-truncation',
     '-fno-math-errno',
