diff -uNr mesa/src/glx/glxcmds.c mesa.mod/src/glx/glxcmds.c
--- mesa/src/glx/glxcmds.c	2021-09-24 02:01:52.893960947 +0000
+++ mesa.mod/src/glx/glxcmds.c	2021-09-24 10:15:16.575439294 +0000
@@ -2101,6 +2101,7 @@
 		int32_t * numerator, int32_t * denominator)
 {
 #if !defined(GLX_USE_WINDOWSGL)
+#ifndef __ANDROID__
    XF86VidModeModeLine mode_line;
    int dot_clock;
    int i;
@@ -2148,6 +2149,8 @@
       return True;
    }
 #endif
+   return True;
+#endif
 
    return False;
 }
