diff -uNr mesa/src/gallium/targets/dri/target.c mesa.mod/src/gallium/targets/dri/target.c
--- mesa/src/gallium/targets/dri/target.c	2021-09-24 02:02:59.224442197 +0000
+++ mesa.mod/src/gallium/targets/dri/target.c	2021-09-25 07:55:26.081085539 +0000
@@ -2,6 +2,7 @@
 #include "target-helpers/sw_helper.h"
 
 #include "dri_screen.h"
+#include "dri2.c"
 
 #define DEFINE_LOADER_DRM_ENTRYPOINT(drivername)                          \
 const __DRIextension **__driDriverGetExtensions_##drivername(void);       \
