diff -uNr libepoxy/src/dispatch_common.c libepoxy.mod/src/dispatch_common.c
--- libepoxy/src/dispatch_common.c	2021-09-22 02:18:29.570826430 +0000
+++ libepoxy.mod/src/dispatch_common.c	2021-09-22 02:21:48.910895613 +0000
@@ -178,7 +178,7 @@
 #define OPENGL_LIB "/System/Library/Frameworks/OpenGL.framework/Versions/Current/OpenGL"
 #define GLES1_LIB "libGLESv1_CM.so"
 #define GLES2_LIB "libGLESv2.so"
-#elif defined(__ANDROID__)
+#elif false
 #define GLX_LIB "libGLESv2.so"
 #define EGL_LIB "libEGL.so"
 #define GLES1_LIB "libGLESv1_CM.so"
